Lib/test/test_capi/test_pyrun.py

```diff
@@ -0,0 +1,57 @@
+import unittest, os
+from collections import UserDict
+from test.support import import_helper
+from test.support.os_helper import unlink, TESTFN, TESTFN_UNDECODABLE
+
+NULL = None
+_testcapi = import_helper.import_module('_testcapi')
+Py_single_input = _testcapi.Py_single_input
+Py_file_input = _testcapi.Py_file_input
+Py_eval_input = _testcapi.Py_eval_input
+
+
+class PyRunTest(unittest.TestCase):
+
+    # run_fileexflags args:
+    # filename -- filename to run script from
+    # start -- Py_single_input, Py_file_input, Py_eval_input
+    # globals -- dict
+    # locals -- mapping (dict)
+    # closeit -- whether close file or not true/false
+    # cf_flags -- compile flags bitmap
+    # cf_feature_version -- compile flags feature version
+    def test_pyrun_fileexflags(self):
+        def run(*args, filename=TESTFN):
+            return _testcapi.run_fileexflags(filename, Py_file_input, *args)
+
+        with open(TESTFN, 'w') as fp:
+            fp.write("a\n")
+        self.addCleanup(unlink, TESTFN)
+
+        self.assertIsNone(run(dict(a=1)))
+        self.assertIsNone(run(dict(a=1), {}))
+        self.assertIsNone(run({}, dict(a=1)))
+        self.assertIsNone(run({}, UserDict(a=1)))
+        self.assertIsNone(run(dict(a=1), {}, 1))  # closeit = True
+
+        if TESTFN_UNDECODABLE is not None:
+            with open(TESTFN_UNDECODABLE, 'w') as fp:
+                fp.write("b\n")
+            self.addCleanup(unlink, TESTFN_UNDECODABLE)
+            self.assertIsNone(run(dict(b=1), filename=TESTFN_UNDECODABLE))
```

Comment:
@serhiy-storchaka, I think you need to separate this into its own test function. These paths do not work for a lot of platforms, for example WASI or any Apple platform.